Tests/NIOTestUtilsTests/NIOHTTP1TestServerTest.swift

```diff
@@ -458,6 +458,55 @@ class NIOHTTP1TestServerTest: XCTestCase {
         XCTAssertNotNil(channel)
         XCTAssertNoThrow(try channel.closeFuture.wait())
     }
+
+    func testCloseChannelWhileItIsWaiting() throws {
+        let testServer = NIOHTTP1TestServer(group: self.group, aggregateBody: false)
+        let firstResponsePromise = self.group.next().makePromise(of: String.self)
+        let firstChannel = try self.connect(serverPort: testServer.serverPort, responsePromise: firstResponsePromise)
+            .wait()
+
+        // Send a request head and wait for it to be sent, and received at the test server so we know the connection is well underway.
+        let requestHead = HTTPRequestHead(version: .http1_1, method: .POST, uri: "/uri")
+        firstChannel.writeAndFlush(SendableRequestPart.head(requestHead), promise: nil)
+        XCTAssertNoThrow(
+            try testServer.receiveHeadAndVerify { head in
+                XCTAssertEqual(head.uri, "/uri")
+            }
+        )
+
+        // Create a second channel now, and again send a request head. We can't wait for the test server to receive it, because it hasn't yet.
```

Comment:
We're not sending a request head here, are we?

Reply:
Good catch, no.

Reply:
Handled.